scripts/support/runner/intel_s1000.py

```diff
@@ -0,0 +1,138 @@
+        parser.add_argument('--ocd-jtag-instr', default='dsp0_gdb.txt',
+                            help='ocd-jtag-instr, defaults to dsp0_gdb.txt')
+        parser.add_argument('--gdb-flash-file', default='load_elf.txt',
+                            help='gdb-flash-file, defaults to load_elf.txt')
+
+    @classmethod
+    def create_from_args(command, args):
+        return intel_s1000BinaryRunner(
+            args.board_dir, args.xt_ocd_dir,
+            args.ocd_topology, args.ocd_jtag_instr, args.gdb_flash_file,
+            args.kernel_elf, args.gdb,
+            gdb_port=args.gdb_port, debug=args.verbose)
+
+    def do_run(self, command, **kwargs):
+        kwargs['ocd-topology'] =  path.join(self.board_dir, 'support',  self.ocd_topology)
+        kwargs['ocd-jtag-instr'] = path.join(self.board_dir, 'support', self.ocd_jtag_instr)
+        kwargs['gdb-flash-file'] = path.join(self.board_dir, 'support', self.gdb_flash_file)
+
+        if command == 'flash':
+            self.flash(**kwargs)
+        elif command == 'debugserver':
+            self.debugserver(**kwargs)
+        else:
+            self.do_debug()
+
+    def flash(self, **kwargs):
+        topology_file = kwargs['ocd-topology']
+        jtag_instr_file = kwargs['ocd-jtag-instr']
+        gdb_flash_file = kwargs['gdb-flash-file']
+
+        self.print_gdbserver_message(self.gdb_port)
+        server_cmd = [self.xt_ocd_dir,
+                        '-c', topology_file,
+                        '-I', jtag_instr_file]
+
+        # Start the server
+        # Note that XTOCD always fails the first time. It has to be
+        # relaunched the second time to work.
+        self.popen_ignore_int(server_cmd)
+        time.sleep(3)
+        server_proc = self.popen_ignore_int(server_cmd)
+        time.sleep(3)
+
+        # Start the client
+        gdb_cmd = [self.gdb_cmd, '-x', gdb_flash_file]
+        client_proc = self.popen_ignore_int(gdb_cmd)
+
+        # Wait for 3 seconds (waiting for XTGDB to finish)
+        time.sleep(3)
+
+        # At this point, the ELF image is loaded and the program is in
+        # execution. Now we can quit the client (xt-gdb) and the server
+        # (xt-ocd) as they are not needed anymore. The loaded program
+        # (ELF) will continue to run though.
+        client_proc.terminate()
+        server_proc.terminate()
+
+    def do_debug(self):
+        if self.elf_name is None:
+            raise ValueError('Cannot debug; elf is missing')
+        if self.gdb_cmd is None:
+            raise ValueError('Cannot debug; no gdb specified')
+
+        gdb_cmd = [self.gdb_cmd,
+                    '-ex', 'target remote :{}'.format(self.gdb_port),
+                    self.elf_name]
+
+        # The below statement will consume the "^C" keypress ensuring
+        # the python main application doesn't exit. This is important
+        # since ^C in gdb means a "halt" operation.
```

Comment:
that's what the popen_ignore_int call is meant to handle, except it is cross-platform, unlike this call which only works on POSIX systems... why use that above but not here?

Reply:
If i use popen_ignore_int call here, the behaviour becomes extremely weird. if i press Enter, it alternates between gdb and the shell. That's why i had to come up with this thing.